### ISSUE TITLE
Terraform lockfiles - be able to specify platform

### ISSUE THREAD

**Issue by tommoyang** [renovatebot/renovate#11632]
### What would you like Renovate to be able to do?

Renovate currently seems to retrieve _all_ platforms for a given provider version. This can be disruptive if you've used `terraform providers lock --platform=*` to only generate hashes for specific platforms - Renovate adds all of the other platform hashes to the list, and we had to remove them manually elsewhere.

```diff
 provider "registry.terraform.io/datadog/datadog" {
-  version     = "2.26.1"
-  constraints = "~> 2.26.0"
+  version     = "3.3.0"
+  constraints = "~> 3.3.0"
   hashes = [
-    "h1:SiCDnaUq1ACnbPGQvdMoqkAK8AtNjQfnnID7pXbsREs=",
-    "h1:d9BGBe8Cy4XowE+5WkHwq8WV2zrIRsfaOXkLwLuHd9M=",
-    "zh:0227ca35de64411b49500a2cd01a02b26fc7d849e057454ccab260fa32110755",
-    "zh:052a052fb1dff2a52ba51ae1d4f69d02d339fc5d5472161172843129a04112b2",
-    "zh:0b9aab6786ea2d7397ec0ca98e2e3d87cc63919aaa373e939b30a9872b5b28bd",
-    "zh:24f3936ea45c46719864252febb506e6f76c0bbbfc47e1a99dbb5baae48c98f3",
-    "zh:25c8f8bd56549477554826288a42a6b814c3920b8e2e19094a724ecb102152ca",
-    "zh:37be57aa93f37d64187f2c214a26856cce8dfafe128d51ba2cedc66198191621",
-    "zh:3a699a41e909fc661b3c3242149d471e0ac37377df8a69edffc5ccf3ff0fc5a6",
-    "zh:3e06d1c8802ff4413c6eaf79cd51a87275db3e41f2bd644640fc666e1c012d32",
-    "zh:8ba201cbd22715bac9bd1dbd41a02196ecf0e6e80bc9d7cdc7bbc9b1e15ffa65",
-    "zh:a395ec039b3896d5d957f606ff22f2e68875ffd86ed8d26d48b4dfa8799aab0a",
-    "zh:b9ec11c6f248247333e097a9b48096b9556dfe8f0bb2d32b4c6229926f0549b5",
-    "zh:bcfc6481a09e47e3eeef0f6df689b9485947706b7198f8f2341e06156d43bbfe",
-    "zh:d014ddd380142bc36ab8b5365b6daf7181350759d979832d489a46aae88dbe9b",
+    "h1:/y4u2dOCzb7vVS+K+C5NF7QYude/t7in30AkPf3mOHw=",
+    "h1:5EUgCTFTNSkii2s/vWCQR6wKt4CYl/SYiOLtNrIRFKw=",
+    "h1:8S5ouF03xaTPywy5WM8Nkn+pLdsSWr2BugRN86q7SsY=",
+    "h1:9m4lPbOeP/jixMMQWEx2TKsvvzAtVYifoNimBNhB3nc=",
+    "h1:J326cfgsNZPF5lK3dtHH2VakNte43CR2VZFG9/iIPJo=",
+    "h1:S7fR/xNLH0PuQmPonxPY4LucSGT7whkRgf4ybKyilTs=",
+    "h1:dx/zoDuBiiurjUTF3S63fp4VnJcdkFJpojS1zi/f2M8=",
+    "h1:fl1EIWBOk12ZxVomKXvHO/VV0yOMEjgUTKpLWnwNc+s=",
+    "h1:inGiB4eFiQq9FiMGUoSGEHSIBuz/nJmK4r3gDx/CkLw=",
+    "h1:npxdGhq2qF5yMd6rsV8E7bCr3/CrFLvmtdEcEp3LHho=",
+    "h1:rV98Yqa2ri7a1V0bIl8Yh9Hu9QV5U05WZHIhIw73JXc=",
+    "h1:t2vvMCaAK/JHFxSAfkAf4XKdGBkEy0NDflEzOc8N5gU=",
+    "h1:u5fSs/e6OFmGut1bCwlA9GGHQfy/AMx2HvioH52M2ck=",
   ]
 }
```

Is it possible to filter the platforms Renovate generates `h1` hashes for, to prevent adding hashes for non-target platforms?

### If you have any ideas on how this should be implemented, please tell us here.

Possibly passing a list of platforms through to https://github.com/renovatebot/renovate/blob/509a3999ecebfa2a1572444af2e09056078936a5/lib/datasource/terraform-provider/index.ts#L195-L220

### Is this a feature you are interested in implementing yourself?

No, mostly because i'm not familiar enough with Renovate to know what i'm doing

**Comment by viceice** [comment]
How can / should renovate detect this? It's not possible to reverse the hashes.

What's your problem with the additional hashes?

**Comment by tommoyang** [comment]
In the changelogs I looked at, Renovate had made large changes to the hash list, caused by adding all of them. I was thinking that being able to specify hashes would reduce the changes made.

That said, it turns out I was mistaken about why we limited hashes (it was to reduce the time taken when manually updating the lockfile, so that terraform only spent time pulling providers we were actually using). The current behaviour of Renovate is fine.

Sorry about the noise! Going to close this issue.
